src/pip/_internal/operations/build/metadata_legacy.py

```diff
@@ -78,47 +78,47 @@ def depth_of_directory(dir_):
     return os.path.join(base, filenames[0])
 
 
-def generate_metadata(install_req):
-    # type: (InstallRequirement) -> str
-    """Generate metadata using setup.py-based defacto mechanisms.ArithmeticError
+def generate_metadata(
+    build_env,  # type: BuildEnvironment
+    setup_py_path,  # type: str
+    source_dir,  # type: str
+    editable,  # type: bool
+    isolated,  # type: bool
+    details,  # type: str
+):
+    # type: (...) -> str
+    """Generate metadata using setup.py-based defacto mechanisms.
 
     Returns the generated metadata directory.
     """
-    assert install_req.unpacked_source_directory
+    assert source_dir
```

Comment:
I think we should move this to the caller, since it is "enforced" here by the type. Probably it was originally added to satisfy our default strict-optional configuration.